precompile/modules/initia_stdlib/sources/dex.move

```diff
@@ -27,6 +27,20 @@ module initia_std::dex {
         coin_b_store: Object<FungibleStore>
     }
 
+    struct FlashSwapLock has key {
+        coin_a_borrow_amount: u64,
+        coin_b_borrow_amount: u64
+    }
+
+    /// FlashSwapReceipt is following the hot potato pattern, so the borrower
+    /// need to pass the created FlashSwapReceipt object to this contract to
+    /// destruct it.
+    ///
+    /// https://move-book.com/programmability/hot-potato-pattern.html
+    struct FlashSwapReceipt {
```

Comment:
another way of doing it (that breaks backward compatibility) is:
1. add `locked` to `struct Pool`
2. add `coin_a_amount` and `coin_b_amount` to `FlashSwapReceipt` (if you agree that `pool_amounts` is not reachable)

but i understand we want to maintain backward compatibility for now, so lgtm